File: apps/ios/Barns/BarnsApp/Domain/UseCases/Catalog/GetCatalogItemsUseCase.swift
/// Returns the official, read-only catalog items.
struct GetCatalogItemsUseCase {
    private let repository: CatalogRepository

    init(repository: CatalogRepository) {
        self.repository = repository
    }

    func execute() async throws -> [CatalogItem] {
        try await repository.catalogItems()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/catalog/GetCatalogItemsUseCase.kt
package com.barns.app.domain.usecase.catalog

import com.barns.app.domain.model.CatalogItem
import com.barns.app.domain.repository.CatalogRepository

/** Returns the official, read-only catalog items. */
class GetCatalogItemsUseCase(
    private val repository: CatalogRepository,
) {
    suspend fun execute(): List<CatalogItem> = repository.catalogItems()
}
